### PR TITLE
Feature/allow registering to events

### PR DESCRIPTION
**Pull request by alextremp**
## Description

This PR adds an optional parameter to BorosTcf initializer in order to pass a `reporter` function to which any relevant events of the consent lifecycle will be emitted. It can be used to debug, collect usage metrics, etc.

## Solves ticket/s

* https://jira.scmspain.com/browse/PSP-3594

## Expected behavior

* If a reporter is not given, nothing changes
* If a reporter is given, it'll be called every time an event is raised to the DomainEventBus

## Review steps

* can be checked manually in local env, the local demo file has been updated
`npm run start`

## Memetized description

mandatory

### CODE REVIEW

File: src/main/domain/status/Status.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ export class Status {
   /**
    *
    * @param {Object} param
+   * @param {DomainEventBus} param.domainEventBus
```
Comment (maryammdot) [inline src/main/domain/status/Status.js:52]: Why do you add this comment here if DomainEventBus is not a param in this class?

File: src/webpack/local.index.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import BorosTcf from '../main'
 
-BorosTcf.init()
+const reporter = {
```
Comment (maryammdot) [inline src/webpack/local.index.js:3]: reporter shouldn't be only a function? not an object with method notify

**Review by maryammdot — APPROVED** [review]

File: src/main/domain/status/Status.js (hunk 1 above)
Comment (alextremp) [reply src/main/domain/status/Status.js:52]: ups, I left it here from a removed probe, removing it thanks!

File: src/webpack/local.index.js (hunk 2 above)
Comment (alextremp) [reply src/webpack/local.index.js:3]: ups, I've changed to a function in a last commit and didn't test the local demo, thanks!

**Pull request merged**